.travis.yml

```diff
@@ -1,18 +1,25 @@
-sudo: required 
-dist: trusty 
+sudo: required
+dist: trusty
 language: generic
 notifications:
   email:
     on_success: change
     on_failure: always
 env:
   global:
-    - ROS_DISTRO="indigo"
     - UPSTREAM_WORKSPACE=file
-    - ROSINSTALL_FILENAME=.travis.rosinstall
   matrix:
-    - ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+    - ROS_DISTRO="indigo" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+    #- ROS_DISTRO="indigo" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+    - ROS_DISTRO="jade" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+    #- ROS_DISTRO="jade" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+    - ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
+    #- ROS_DISTRO="kinetic" ROS_REPOSITORY_PATH=http://packages.ros.org/ros-shadow-fixed/ubuntu
+matrix:
+  allow_failures:
+    - env: ROS_DISTRO="jade" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu UPSTREAM_WORKSPACE=file
```

Comment:
>  No need to define UPSTREAM_WORKSPACE=file in this job because it's already globally defined.

Travis expects all variable (?) for matching `allow_failures` (https://docs.travis-ci.com/user/customizing-the-build#Rows-that-are-Allowed-to-Fail)